docs/docs/consumer.mdx

```diff
@@ -35,45 +35,58 @@ run any analytics you may wish to perform across the whole model.
+    GrpcResult result = client.getEquipmentContainer("container_mrid").throwOnError();
+    NetworkService ns = client.getService();
 }
 
-// With SSL and OAuth2. Note that you should never connect to a non-SSL server while using OAuth2, as you will be vulnerable to MITM attacks.
-try (GrpcChannel channel = GrpcChannelFactory.create(new ConnectionConfig("localhost", 50051, true, "ca.cert"))) {}
-    // Create CallCredentials using a configuration URL. This configuration endpoint will often be provided by the service you are connecting to.
-    JwtCredentials creds = EvolveCallCredentials.create(
-        "<client_id>",
-        "<client_secret>",
-        "<ewb_auth_config_URL>"
-    );
-
-    // Or create them with an AuthConfig if a configuration endpoint doesn't exist.
-    JwtCredentials creds = EvolveCallCredentials.create(
-        "<client_id>",
-        "<client_secret>",
-        new AuthConfig(tokenUrl="https://zepben.au.auth0.com/oauth/token", audience="https://evolve/", authType=AuthType.AUTH0)
-    );
-
-    NetworkConsumerClient client = new NetworkConsumerClient(channel, creds);
-    NetworkService ns = new NetworkService();
-    GrpcResult result = client.getFeeder(ns).throwOnError();
+// With SSL and OAuth2
+try (GrpcChannel channel = Connect.connectWithSecret(
+    "client_id",
+    "client_secret",
+    "config_address", // set to null to fetch configuration from "https://{rpc hostname}/ewb/auth"
+    "config_CA_filename", // set to null to use system CAs for fetching configuration
+    "auth_CA_filename", // set to null to use system CAs for fetching access tokens
+    "localhost", // rpc hostname
+    50051, // rpc port
+    new File("path/to/zepben/ca.cert"))
+) {
+    NetworkConsumerClient client = new NetworkConsumerClient(channel);
+    GrpcResult result = client.getEquipmentContainer("container_mrid").throwOnError();
+    NetworkService ns = client.getService();
+}
+
+// Or specify the authentication configuration directly:
+try (GrpcChannel channel = Connect.connectWithSecret(
+    "client_id",
+    "client_secret",
+    "https://evolve-ewb/",
+    "zepben.au.auth0.com",
+    AuthMethod.Auth0,
+    "auth_CA_filename", // set to null to use system CAs for fetching access tokens
+    "localhost", // rpc hostname
+    50051, // rpc port
+    new File("path/to/zepben/ca.cert"))
+) {
+    NetworkConsumerClient client = new NetworkConsumerClient(channel);
+    GrpcResult result = client.getEquipmentContainer("container_mrid").throwOnError();
+    NetworkService ns = client.getService();
 }
 
 // With SSL and client authentication
-try (GrpcChannel channel = GrpcChannelFactory.create(new ConnectionConfig("localhost", 50051, true, "ca.cert", "path/to/signed.cert", "path/to/private.key"))) {
+try (GrpcChannel channel = new GrpcChannelBuilder().forAddress("localhost", 50051)
+    .makeSecure(new File("ca.cert"), new File("certchain.pem"), new File("private_key")).build()) {
```

Comment:
Maybe we should use filenames instead of `java.io.File`. The only reason I went with Files in the first place was to match the Python sdk.

Reply:
Yeah happy to use filenames, would be much cleaner